### ISSUE TITLE
Salvattore *almost* ported to Backdrop...

### ISSUE THREAD

**Issue by bserem** [backdrop-ops/contrib#70]
Well, the last one was the most tricky one.

Salvattore is a masonry grid that gets configured through CSS. (http://salvattore.com)
Yesterday I succesfully managed to create a module for Drupal for it, my first attempt to create a views style module.

Sadly, I can't do the same for Backdrop.
The backdrop version is here:
https://github.com/bserem/salvattore (v1.x-1.x)
and the diff from D7 is:
https://github.com/bserem/salvattore/blob/1.x-1.x/changes_from_d7.diff

The settings form works. I can't get Views to detect the new style.
After that, adding the CSS and JS file should be fairly easy (just rename a couple of functions and change to config_get).
If anybody has any idea on how to register the style in views we will be having the first masonry plugin for Backdrop :)

**Comment by quicksketch** [comment]
Hi @bserem! Awesome! Thanks for porting this module to Backdrop! It's odd that it's not showing up. The Views API is nearly identical. The biggest change you should have is that the autoloader changed from being in the .info file to being hook_autoload_info(), and you included that. I'll have to test this out locally and see if I can figure out the cause of the problem here.

**Comment by bserem** [comment]
I've found out a typo on my Drupal version of the module. But it was on the .tpl.php file. I don't think this is the reason it is not showing up.
It is this commit https://github.com/bserem/salvattore/commit/76e64d9e8833659473b0fe548c7a02f9bc088396#diff-3396a289d4629f67f6f47629962b52eaR14 (row 14 of tpl file).

I'm missing something obviously. Will have a look again but I'm officially stuck at the moment.
Hope you manage to solve it ;)

**Comment by quicksketch** [comment]
Looks like @docwilmot figured this one out in https://github.com/backdrop/backdrop-issues/issues/836#issuecomment-87935836.

**Comment by bserem** [comment]
Yes, I'm communicating with him, once we prepare this we'll put this on backdrop contrib.
Thanks

**Comment by klonos** [comment]
@bserem: Geia sou re patrida! Thanx for creating the Drupal module and for porting it to Backdrop!

Another idea for a port is the [Equal Heights jQuery](https://www.drupal.org/project/equalheights) that uses [jQuery Equal Heights plugin](http://www.cssnewbie.com/equalheights-jquery-plugin/) to make elements have equal heights. You can either port the module itself to Backdrop from the Drupal one, or incorporate the jQuery plugin in Salvatore for added functionality (a "Make grid elements have equal height" checkbox).

**Comment by klonos** [comment]
...BTW, this is another example of a module that uses an external library. The 3rd party library is placed in a ```js``` subfolder within the module folder tree, whereas it should be placed in a ```libraries``` subfolder and in fact under root similar to ```\modules``` and ```\themes``` (leaving the ```js``` subfolder for custom, "private" js, only applicable to the module itself). We need to decide on and implement that now and put best-practice rules/suggestions in place before we have too many such modules that use 3rd party libraries.

In other words... Libraries API:
http://drupal.org/project/libraries
https://github.com/backdrop-contrib/libraries
https://github.com/backdrop/backdrop-issues/issues/159
